olmo/train.py

```diff
@@ -763,9 +790,13 @@ def train_step(self, batch: Dict[str, Any], reduce_global_loss: bool = True) ->
             metrics["train/ZLoss"] = z_batch_loss.item()
 
         # Maybe collect post-step optimizer-specific metrics.
+        # TODO (ananya): self.dist_model with DDP will not work with ``get_post_step_metrics`` of LionW
+        # while AdamW does not implement a custom version of ``get_post_step_metrics`` function. Since we do not use
+        # LionW anymore, we can skip fixing the interaction of DDP model and ``get_post_step_metrics`` function
+        # in LionW for the time being.
```

Comment:
Instead of putting this comment here, I would suggest throwing an `assert isinstance(model, FSDP)` in `LionW.get_post_step_metrics()`. That way this fails loudly.